scene/3d/gpu_particles_3d.cpp

```diff
@@ -127,6 +127,7 @@ void GPUParticles3D::set_process_material(const Ref<Material> &p_material) {
 	RID material_rid;
 	if (process_material.is_valid()) {
 		material_rid = process_material->get_rid();
+		process_material->connect("emission_shape_changed", callable_mp((Node3D *)this, &GPUParticles3D::update_gizmos));
```

Comment:
This doesn't check that `process_material` is a `ParticlesProcessMaterial`, so it fails when it's a `ShaderMaterial`.

Reply:
@akien-mga Fix PR is up: https://github.com/godotengine/godot/pull/101617